transports/uds/src/lib.rs

```diff
@@ -211,13 +211,15 @@ mod tests {
                         .map_err(|err| panic!("IO error {:?}", err));
 
                     // Spawn the future as a concurrent task
-                    tokio_current_thread::spawn(handle_conn);
-
+                    let mut rt = Runtime::new().unwrap();
+                    let _ = rt.spawn(handle_conn);
+                    std::thread::sleep(std::time::Duration::from_millis(100));
```

Comment:
There's something sketchy going on if this line is needed for the test to pass.

Reply:
There is something sketch going on for sure. I seem to get a broken pipe from the test sporadically without the sleep. I will dig into it later and try to get a more suitable fix. ( the other sleeps in this test are smelly too imo ).
I assume it is because the reader doesn't wait for the read to be complete before closing the socket. I would block/wait on the future but I'm not sure it that is in the spirit of the test as it was clearly intending to do an async operation.